src/EnergyPlus/EconomicTariff.cc

```diff
@@ -224,32 +224,41 @@ void GetInputEconomicsTariff(EnergyPlusData &state, bool &ErrorsFound) // true i
         tariff(iInObj).kindGasMtr = kindMeterNotGas;
 
         // Determine whether this meter is related to electricity, or water, or gas
-        if (tariff(iInObj).reportMeterIndx != 0) {
-
-            std::string const &SELECT_CASE_var(Util::makeUPPER(state.dataOutputProcessor->EnergyMeters(tariff(iInObj).reportMeterIndx).ResourceType));
-
+        if (tariff(iInObj).reportMeterIndx != -1) {
+            switch (state.dataOutputProcessor->meters[tariff(iInObj).reportMeterIndx]->resource) {
             // Various types of electricity meters
-            if (SELECT_CASE_var == "ELECTRICITY") {
+            case Constant::eResource::Electricity: {
                 tariff(iInObj).kindElectricMtr = kindMeterElecSimple;
-            } else if (SELECT_CASE_var == "ELECTRICITYPRODUCED") {
+            } break;
+            case Constant::eResource::ElectricityProduced: {
                 tariff(iInObj).kindElectricMtr = kindMeterElecProduced;
-            } else if (SELECT_CASE_var == "ELECTRICITYPURCHASED") {
+            } break;
+            case Constant::eResource::ElectricityPurchased: {
                 tariff(iInObj).kindElectricMtr = kindMeterElecPurchased;
-            } else if (SELECT_CASE_var == "ELECTRICITYSURPLUSSOLD") {
+            } break;
+            case Constant::eResource::ElectricitySurplusSold: {
                 tariff(iInObj).kindElectricMtr = kindMeterElecSurplusSold;
-            } else if (SELECT_CASE_var == "ELECTRICITYNET") {
+            } break;
+            case Constant::eResource::ElectricityNet: {
                 tariff(iInObj).kindElectricMtr = kindMeterElecNet;
-
-                // Handle the case where its a water meter
-            } else if (SELECT_CASE_var == "WATER" || SELECT_CASE_var == "H2O" || SELECT_CASE_var == "ONSITEWATER" ||
-                       SELECT_CASE_var == "WATERPRODUCED" || SELECT_CASE_var == "ONSITE WATER" || SELECT_CASE_var == "MAINSWATER" ||
-                       SELECT_CASE_var == "WATERSUPPLY" || SELECT_CASE_var == "RAINWATER" || SELECT_CASE_var == "PRECIPITATION" ||
-                       SELECT_CASE_var == "WELLWATER" || SELECT_CASE_var == "GROUNDWATER" || SELECT_CASE_var == "CONDENSATE") {
```

Comment:
I presume some of these were unused.

Reply:
I don't know what you mean. We are using enums internally now so all of the variants I found internally were just replaced by the same enum as appropriate. Are you saying that some of these come in through the IDF file and would need to be handled explicitly as strings? I didn't really see any paths like that in the code and nothing came up in the test suite ...

Reply:
I noticed that "H2O" was one of them, which may be just a remnant from the past.